h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -128,6 +128,23 @@ protected ModelBuilder(P parms, boolean startup_once, String externalSchemaDirec
   /** gbm -> "hex.schemas." ; custAlgo -> "org.myOrg.schemas." */
   public static String schemaDirectory(String urlName) { return SCHEMAS[ArrayUtils.find(ALGOBASES,urlName)]; }
 
+  /**
+   * Helper method that enables preprocessing with TE model. It's user's responsibility to make sure TE is trained in an appropriate way to avoid data leakage.
+   * E.g. TE without KFold strategy being applied for cv main model leads to a data leakage.
+   * @param model key to the trained TE model
+   */
+  public void addTEModelKey(Key<Model> model) {
+    _parms._te_model = model;
+  }
+
+  /**
+   *
+   * @return TE model's key if one was assigned to a model builder, null otherwise
+   */
+  public Key<Model> internal_getTEModelKey() {
```

Comment:
If something is internal, let's not have it public and use packages to moderate the access. Also, Java is not snake_case.

Reply:
verified that it was not needed for AutoML integration, so, visibility should definitely be changed.